src/sd_jwt.rs

```diff
@@ -86,15 +189,281 @@ impl FromStr for SdJwt {
   }
 }
 
+#[derive(Debug, Clone)]
+pub struct SdJwtPresentationBuilder {
+  sd_jwt: SdJwt,
+  disclosures: HashMap<String, Disclosure>,
+  removed_disclosures: Vec<Disclosure>,
+  object: Value,
+}
+
+impl Deref for SdJwtPresentationBuilder {
+  type Target = SdJwt;
+  fn deref(&self) -> &Self::Target {
+    &self.sd_jwt
+  }
+}
+
+impl SdJwtPresentationBuilder {
+  pub fn new(mut sd_jwt: SdJwt, hasher: &dyn Hasher) -> Result<Self> {
+    let required_hasher = sd_jwt.claims()._sd_alg.as_deref().unwrap_or(SHA_ALG_NAME);
+    if required_hasher != hasher.alg_name() {
+      return Err(Error::InvalidHasher(format!(
+        "hasher \"{}\" was provided, but \"{required_hasher} is required\"",
+        hasher.alg_name()
+      )));
+    }
+    let disclosures = std::mem::take(&mut sd_jwt.disclosures)
+      .into_iter()
+      .map(|disclosure| (hasher.encoded_digest(disclosure.as_str()), disclosure))
+      .collect();
```

Comment:
While trying out your other SD-JWT related PR (https://github.com/iotaledger/identity.rs/pull/1413) I ran into the problem that while trying to validate the key binding of a 'freshly' created SD-JWT VC `validate_key_binding` would sometimes randomly fail on [this assertion](https://github.com/iotaledger/identity.rs/blob/a510185d80edfef91a72d675962e23eebb068ab0/identity_credential/src/sd_jwt_vc/token.rs#L372-L374):
```rust
    if kb_jwt.claims().sd_hash != digest {
      return Err(Error::Validation(anyhow!("invalid KB-JWT's `sd_hash`")));
    }
```

I managed to pin the reason down to this piece of code since the disclosures are being collected into a `HashMap` here and therefore being scrambled. That is not necessarily a problem since the spec actually states that it is [RECOMMENDED to shuffle the array of hashes](https://github.com/iotaledger/identity.rs/blob/a510185d80edfef91a72d675962e23eebb068ab0/identity_credential/src/sd_jwt_vc/token.rs#L372-L374) (of the digests). However, the problem is that at this point in the code, the `_sd_hash` has already been generated before the scrambling of the digests which causes `validate_key_binding` to fail at random (because the randomly changed order of the digests causes the `digest` to differ from the `_sd_hash` in `validate_key_binding`).

I could easily fix the problem by changing the `disclosures` type to an `IndexMap` in order to preserve the order of the disclosures:
```rust
#[derive(Debug, Clone)]
pub struct SdJwtPresentationBuilder {
  sd_jwt: SdJwt,
  disclosures: IndexMap<String, Disclosure>,
  removed_disclosures: Vec<Disclosure>,
  object: Value,
}
```

Reply:
@nanderstabel thank you for trying this out. This was definitely something I overlooked. I'll update the code with your suggestion